deploy-board/deploy_board/webapp/cluster_view.py

```diff
@@ -106,14 +107,19 @@ def post(self, request, name, stage):
                 request, name, stage, capacity_type="GROUP", data=cluster_name)
 
             clusters_helper.create_cluster_with_env(request, cluster_name, name, stage, cluster_info)
-
-            return HttpResponse("{}", content_type="application/json")
         except NotAuthorizedException as e:
             log.error("Have an NotAuthorizedException error {}".format(e))
-            return HttpResponse(e, status=403, content_type="application/json")
+            ret = 403
         except Exception as e:
             log.error("Have an error {}".format(e))
-            return HttpResponse(e, status=500, content_type="application/json")
+            ret = 500
+        finally:
+            if ret == 200:
+                return HttpResponse("{}", content_type="application/json")
+            else:
+                environs_helper.remove_env_capacity(
+                    request, name, stage, capacity_type="GROUP", data=cluster_name)
+                return HttpResponse(e, status=ret, content_type="application/json")
```

Comment:
Is the exception `e` available in this scope?

Reply:
yes